pkg/agent/plugin/workloadattestor/docker/cgroup/dockerfinder.go

```diff
@@ -0,0 +1,103 @@
+package cgroup
+
+import (
+	"fmt"
+	"regexp"
+	"strings"
+)
+
+// Token is a character or set of characters that are used during parsing of a cgroup entry.
+type Token string
+
+const (
+	// WildcardToken is used to match a variable number of any character, excluding forward slash (`/`).
+	//
+	// This token MAY be provided any number of times.
+	WildcardToken Token = "*"
+
+	// ContainerIDToken is used to match the container id, which MUST not include a forward slash (`/`).
+	//
+	// This token MUST be provided exactly once in a pattern.
+	ContainerIDToken Token = "<id>"
+
+	// used for converting from our tokens to those understood by the regexp package
+	regexpWildcard         = "[^\\/]*"
+	regexpWildcardSubmatch = "([^\\/]*)"
+
+	// index for slice returned by FindSubmatch
+	// fullmatchIndex = 0
+	submatchIndex = 1
+)
+
+// ContainerIDFinder finds a container id from a cgroup entry.
+type ContainerIDFinder interface {
+	// FindContainerID returns a container id and true if the known pattern is matched, false otherwise.
+	FindContainerID(cgroup string) (containerID string, found bool)
+}
+
+func newContainerIDFinder(pattern string) (ContainerIDFinder, error) {
```

Comment:
Should this allow for a pattern like `/asdf<id>` or is the `<id>` token always intended to fill an entire segment?

Reply:
I think it should fill an entire segment. I don't think the docker cgroupParent will allow anything else to happen.

Reply:
I only asked because the k8s workload attestor handles prefixes to that segment (i.e. `docker-`, `crio-`) as well as suffixes (i.e. `.scope`). Those might only apply to the kubepods entries made be kubernetes though.... it just caught my attention (I have no working knowledge on how these things are formed).

Reply:
https://github.com/moby/moby/blob/dde030a6b16de026d0921d1f107845666eecfb18/daemon/oci_linux.go#L770

This is either a "systemd form" (which this plugin doesn't currently support; I'd like to hold off on supporting that until someone asks) or a "docker" form. The docker form is *always* a `filepath.Join` on the parent so the container ID will always be an entire segment.

Reply:
I see, ok. This seems good for now.